teuthology/suite/run.py

```diff
@@ -564,39 +569,102 @@ def check_num_jobs(self, jobs_to_schedule):
         if threshold and jobs_to_schedule > threshold:
             util.schedule_fail(msg, dry_run=self.args.dry_run)
 
-    def schedule_suite(self):
+    def descr_to_yamls(self, descriptions, suites_path):
         """
-        Schedule the suite-run. Returns the number of jobs scheduled.
+        Function converts description of the run into sequence of .yaml files.
+        Input "descriptions" is the value of "--descr" parameter;
+        "description" consists of comma-separated list of run description
+        Input:
+        descriptions="rerun/rados/objectstore/{backends/objectstore-bluestore-b supported-random-distro$/{centos_latest}}"
+        suites_path="/cephfs/github.com_ceph_build/qa/suites/"
+        Output:
+        ('rados/objectstore/{backends/objectstore-bluestore-b supported-random-distro$/{ubuntu_latest}}',
+        ['/cephfs/github.com_ceph_build/qa/suites/rados/objectstore/backends/objectstore-bluestore-b.yaml',
+         '/cephfs/github.com_ceph_build/qa/suites/rados/objectstore/supported-random-distro$/ubuntu_latest.yaml'])
         """
-        name = self.name
-        if self.args.arch:
-            arch = self.args.arch
-            log.debug("Using '%s' as an arch" % arch)
-        else:
-            arch = util.get_arch(self.base_config.machine_type)
-        suite_name = self.base_config.suite
-        suite_path = os.path.normpath(os.path.join(
+        def expand_descr(test_yamls, source, prefix, base_pos):
```

Comment:
no descriptions for this method? I think since the function returns values and do some test_yaml modifications, it is probably better to describe briefly what is the idea behind and give a better name for it.

Reply:
btw, we already used to use "desc" shortened for description, maybe worth to continue using instead of "descr"

Reply:
Added description.